src/components/screensavermanager.js

```diff
@@ -0,0 +1,133 @@
+define(["events", "playbackManager", "pluginManager", "inputManager", "connectionManager", "userSettings"], function (events, playbackManager, pluginManager, inputManager, connectionManager, userSettings) {
+    "use strict";
+
+    function getMinIdleTime() {
+        // Returns the minimum amount of idle time required before the screen saver can be displayed
+        //time units used Millisecond
+        return 180000;
+    }
+
+    var lastFunctionalEvent = 0;
+
+    function getFunctionalEventIdleTime() {
+        return new Date().getTime() - lastFunctionalEvent;
+    }
+
+    events.on(playbackManager, "playbackstop", function (e, stopInfo) {
+        var state = stopInfo.state;
+        if (state.NowPlayingItem && state.NowPlayingItem.MediaType == "Video") {
+            lastFunctionalEvent = new Date().getTime();
+        }
+    });
+
+    function getScreensaverPlugin(isLoggedIn) {
+
+        var option;
+        var defaultOption = isLoggedIn ? "backdropscreensaver" : "logoscreensaver";
+        try {
+            option = userSettings.get("screensaver", false);
+        } catch (err) {
+            option = null;
+        }
+
+        if (option === "none" || defaultOption === "none") {
+            return null;
+        }
+
+        var plugin = pluginManager.ofType("screensaver").filter(function (i) {
+            return i.id === option;
+        })[0];
```

Comment:
where is error handling of "user wants a screensaver we don't have"?

Reply:
https://github.com/jellyfin/jellyfin-web/pull/592/commits/06b42973f401b5d93b4f692c762145d40cffb6c0